src/ResultWriter/ClusteringWriter.h

```diff
@@ -0,0 +1,39 @@
+#ifndef SEISSOL_CLUSTERINGWRITER_H
+#define SEISSOL_CLUSTERINGWRITER_H
+
+#include <string>
+#include <vector>
+#include <type_traits>
+#include "Initializer/tree/Layer.hpp"
+
+namespace seissol::writer {
+
+class ClusteringWriter {
+public:
+  ClusteringWriter(const std::string& outputPrefix);
+    void addCluster(unsigned profilingId,
+                    unsigned localClusterId,
+                    LayerType layerType,
+                    unsigned size,
+                    unsigned dynRupSize);
+    void write() const;
+
+  // SoA that contains info about clusters
+  struct ClusteringInformation {
+    std::vector<int> ranks;
+    std::vector<int> localRanks;
+    std::vector<int> profilingIds;
+    std::vector<int> localClusterIds;
+    std::vector<std::underlying_type<LayerType>::type> layerTypes;
+    std::vector<unsigned> sizes;
```

Comment:
Just a side comment: will we stay within 32 Bit size mesh/cluster sizes for the foreseeable future? Just wondering about the size of `unsigned` here again...

Reply:
Probably not. In this case it's definitely fine, as we're only dealing with local sizes (and they can't be much more than ~10 million due to memory.